### DISCUSSION TITLE
关于 `matcher.reject` 是否有全局强制中断方法。

### DISCUSSION THREAD

**Discussion by nanshaobit** [nonebot/discussions#43] [问答] [answered]
`matcher.reject` 可以强制接收一条新的消息用来再次运行当前消息处理函数， 此过程中，其他 `on_command` 优先级是不起作用的。
如果用户想退出当前处理函数，有无全局的结束该函数的方法。

例如，我注册了很多 `on_command` 中都有提示用户重新输入的情况，目前只能在每个函数内提示用户 输入q 退出该函数。但是处理函数很多时，这部分代码就显得重复。
我试着注册了一个 `on_command("q", rule=to_me(), priority=1)` 优先级为1 的退出函数，但是在 `reject` 过程中 不起作用。

想了解一下，除了在每个函数内提示用户输入q 并判断，有无更优的解决方案。

多谢～

**Comment by yanyongyu** [comment, marked as answer]
你可以写一个`args_parser`并在其中进行退出判断调用`finish`来进行退出

```python
async def allow_cancel(bot: Bot, event: Event, state: T_State):
    """An args parser allows to finish the session."""
    message = str(event.get_message())
    if message == "取消":
        await bot.send(event, "已取消")
        raise FinishedException
    state[state["_current_key"]] = str(event.get_message())
```

并在matcher中注册

```python
matcher.args_parser(allow_cancel)
```
